xarray/tests/test_variable.py

```diff
@@ -782,14 +784,37 @@ def test_pad(self):
         data = np.arange(4 * 3 * 2).reshape(4, 3, 2)
         v = self.cls(["x", "y", "z"], data)
 
+        # TODO "mean", "median" and "reflect" have issues
+        modes = ["edge", "linear_ramp", "maximum", "minimum", "symmetric", "wrap"]
```

Comment:
I would do this as
```python

@pytest.mark.parametrize("modes", [
pytest.param("mean", marks=pytest.mark.xfail),
pytest.param("median", marks=pytest.mark.xfail),
pytest.param("reflect", marks=pytest.mark.xfail),
"edge", "linear_ramp", "maximum", "minimum", "symmetric", "wrap"])
@pytest.mark.parametrize("xr_args, np_args", 
[
  [{"x": (2, 1)}, ((2,1), (0,0), (0,0))],
...
])
def test_pad(modes, xr_args, np_args):
```

instead of the for loop on line 795

Reply:
Cool, didn't know that was possible!

Reply:
Unfortunately, decorators are not inherited, so I'm not sure about this change. It feels kind of awkward in `TestIndexVariable.test_pad`

Reply:
OK. Leave it for now. I can see if it works and push a change later